amd_tressfx_vulkan/src/TressFXRendererVulkan.cpp

```diff
@@ -0,0 +1,1569 @@
+            getImageMemoryBarrier(m_pNoiseTexture, 0, VK_ACCESS_TRANSFER_WRITE_BIT,
+                                  VK_IMAGE_LAYOUT_UNDEFINED,
+                                  VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL)};
+
+        vkCmdPipelineBarrier(commandBuffer, VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT,
+                             VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT, 0, 0, nullptr, 0,
+                             nullptr, AMD_ARRAY_SIZE(makeNoiseTextureTransferableBarrier),
+                             makeNoiseTextureTransferableBarrier);
+
+        VkBufferImageCopy copyInfo{offsetInScratchBuffer, 512, 512};
+        copyInfo.imageExtent.width = 512;
+        copyInfo.imageExtent.height = 512;
+        copyInfo.imageExtent.depth = 1;
+        copyInfo.imageSubresource.aspectMask = VK_IMAGE_ASPECT_COLOR_BIT;
+        copyInfo.imageSubresource.layerCount = 1;
+        vkCmdCopyBufferToImage(commandBuffer, scratchBuffer, m_pNoiseTexture,
+                               VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL, 1, &copyInfo);
+        offsetInScratchBuffer += 512 * 512 * sizeof(XMFLOAT4);
+
+        VkImageMemoryBarrier makeNoiseTextureSampleableBarrier[] = {getImageMemoryBarrier(
+            m_pNoiseTexture, VK_ACCESS_TRANSFER_WRITE_BIT, VK_ACCESS_SHADER_READ_BIT,
+            VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL,
+            VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL)};
+
+        vkCmdPipelineBarrier(commandBuffer, VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT,
+                             VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT, 0, 0, nullptr, 0,
+                             nullptr, AMD_ARRAY_SIZE(makeNoiseTextureSampleableBarrier),
+                             makeNoiseTextureSampleableBarrier);
+
+        delete[] noiseArray;
+
+        VkImageViewCreateInfo srDesc{VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO};
+        srDesc.format = VK_FORMAT_R32G32B32A32_SFLOAT;
+        srDesc.viewType = VK_IMAGE_VIEW_TYPE_2D;
+        srDesc.subresourceRange.aspectMask = VK_IMAGE_ASPECT_COLOR_BIT;
+        srDesc.subresourceRange.layerCount = 1;
+        srDesc.subresourceRange.levelCount = 1;
+        srDesc.image = m_pNoiseTexture;
+
+        AMD_V_RETURN(vkCreateImageView(pvkDevice, &srDesc, nullptr, &m_pNoiseView));
+    }
+
+    return VK_SUCCESS;
+}
+
+//--------------------------------------------------------------------------------------
+//
+// CreateVertexBuffers
+//
+// Creates the vertex buffers for hair rendering
+//
+//--------------------------------------------------------------------------------------
+VkResult TressFXRenderer::CreateVertexBuffers(
+    VkDevice pvkDevice, uint32_t texture_memory_index, VkCommandBuffer commandBuffer,
+    VkDeviceMemory scratchMemory, VkBuffer scratchBuffer, size_t &offsetInScratchBuffer)
+{
+    VkResult vr;
+    // Create the screen quad vertex buffer(use StandardVertex for simplicity)
+    const StandardVertex screenQuad[6] = {
+        {XMFLOAT3(-1.0f, -1.0f, 0.0f), XMFLOAT3(0.0f, 0.0f, 0.0f),
```

Comment:
Seems like it would be easier to create a screen-covering triangle: https://anteru.net/blog/2012/03/03/1862/